### PR TITLE
glean: 1526343: Add public sendPings API

### PR DESCRIPTION
**Pull request by mdboom**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt [hunk 1]
```diff
@@ -343,7 +374,7 @@ open class GleanInternalAPI internal constructor () {
      * @param pingNames List of pings to send.
      * @return true if any pings were actually sent.
      */
-    internal fun sendPings(pingNames: List<String>): Boolean {
+    internal fun sendPingsInternal(pingNames: List<String>): Boolean {
```
Comment (Dexterp37) [inline components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: nit: this is not strictly related to this patch, but if consumers will start attempting to send empty pings using the `sendPings` API they will receive no feedback if the payload is empty and nothing happens. I'd suggest adding a debug log line [here](https://github.com/mozilla-mobile/android-components/blob/0dfbad94ab623d922d1f9b4d41be6bc594e80f63/components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt#L392) to just say "Nothing to collect for ping: `pingname`".

**Comment by Dexterp37** [comment]
> 
> 
> I'd like to see this documented in the docs as part of this. I don't think it's worth leaving that as technical debt and would illustrate how we intend this to be used and point out what it's not supposed to do, like send glean internal pings.

Oh, great catch!

File: components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt (hunk 1 above)
Comment (mdboom) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: That's not a bad idea.  This information is currently in the return value, too.  Do you think maybe that's not necessary?  None of the uses internally use the return value right now.

Comment (Dexterp37) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: Mh, I'm not sure if that's needed or not. I might have a use case for the `activation` ping, so let's keep it around for now :)

**Pull request merged**